src/settings.js

```diff
@@ -1761,6 +1761,9 @@ var USES_DYNAMIC_ALLOC = 1;
 // library support.
 var SUPPORT_LONGJMP = 1;
 
+// WebAssembly setjmp-longjmp handling using Wasm EH (experimental)
+var SJLJ_HANDLING = 0;
```

Comment:
How about using `SUPPORT_LONGJMP=wasm` which would avoid a new setting and trying to keep them in sync?

Reply:
We could add some code to turn `-s SUPPORT_LONGJMP=1` to write `emscripten`, perhaps?

I am personally a little unsure what is best here. With exceptions we do have two separate options - we didn't try to have a single one for the 3 modes (none, emscripten, wasm). So maybe the PR as it is is more consistent with that. OTOH I do see the benefit of fewer options, and less risk for confusion.

In the long term, I think it would be good to remove emscripten exceptions and sjlj support from LLVM entirely. We can add a lowering pass in binaryen that reduces those features to wasm MVP. And emcc would run that pass automatically when the enabled features determine that. If we can get there, then the options situation would get simpler (also for exceptions). So I don't feel strongly either way I guess about the specifics of the options for now.

Reply:
Yeah it is unfortunately not simple. Also for Emscripten exception there are two options: `DISABLE_EXCEPTION_THROWING` and `DISABLE_EXCEPTION_CATCHING`, so having a single option to control all EH is not easy because of that too.

Here I added `SJLJ_HANLDING` to be consistent with `EXCEPTION_HANDLING`, but it is also not easy as I described, because we have to appropriate error out or decide which one to honor when conflicting options are given. I don't have a strong preference here, but at least using a single option `SUPPORT_LONGJMP` to control all longjmp options looks simpler.

By the way, there are many lines like this in js libraries:
```js
#if SUPPORT_LONGJMP
```
if `SUPPORT_LONGJMP` is a string, can we do things like
```js
#if SUPPORT_LONGJMP == 'emscripten'
```
?

Reply:
Yes, string comparisons in ifdefs should work.

Reply:
I only see a single use of this setting in the library JS files: https://github.com/emscripten-core/emscripten/blob/1267242815c355d7a0b86a7c667790dc4bdcdcda/src/library.js#L1499

And it looks like its doing `if !SUPPORT_LONGJMP` to test if there is no support at all.. which I think will continue to work, no?

Reply:
@sbc100 There's one more usage of `SUPPORT_LONGJMP` in preamble.js: https://github.com/emscripten-core/emscripten/blob/73c20f60cfa046e156a2c3970f7967b00e6662e7/src/preamble.js#L689

And the convention for multiple-option settings in https://github.com/emscripten-core/emscripten/blob/main/src/settings.js seems to something like `optionA`, `optionB`, and `none`; examples are:  https://github.com/emscripten-core/emscripten/blob/73c20f60cfa046e156a2c3970f7967b00e6662e7/src/settings.js#L1933-L1947
https://github.com/emscripten-core/emscripten/blob/73c20f60cfa046e156a2c3970f7967b00e6662e7/src/settings.js#L118-L138
https://github.com/emscripten-core/emscripten/blob/73c20f60cfa046e156a2c3970f7967b00e6662e7/src/settings.js#L1861-L1867

So I think we can make this option take one of the three values: `wasm`, `emscripten`, and `none`. And For backward compatibility, 1 will be converted to `emscripten` and 0 will be converted to `none` in  emcc.py. That's the reason I asked.. If the string comparisons are fine there will be no problem.